classes/search/ArticleSearchIndex.php

```diff
@@ -109,24 +113,28 @@ public function deleteTextIndex($articleId, $type = null, $assocId = null)
     public function submissionFileChanged($articleId, $type, $submissionFile)
     {
         // Check whether a search plug-in jumps in.
-        $hookResult = Hook::call(
-            'ArticleSearchIndex::submissionFileChanged',
-            [$articleId, $type, $submissionFile->getId()]
-        );
+        if (Hook::ABORT === Hook::call('ArticleSearchIndex::submissionFileChanged', [$articleId, $type, $submissionFile->getId()])) {
+            return;
+        }
 
-        // If no search plug-in is activated then fall back to the
-        // default database search implementation.
-        if ($hookResult === false || is_null($hookResult)) {
-            $parser = SearchFileParser::fromFile($submissionFile);
-            if (isset($parser) && $parser->open()) {
-                $searchDao = DAORegistry::getDAO('ArticleSearchDAO'); /** @var ArticleSearchDAO $searchDao */
-                $objectId = $searchDao->insertObject($articleId, $type, $submissionFile->getId());
+        // If no search plug-in is activated then fall back to the default database search implementation.
+        $parser = SearchFileParser::fromFile($submissionFile);
+        if (!$parser?->open()) {
```

Comment:
Hmm, the `try/finally` should enclose this line (I wrote it before seeing the `close()`) 😁